R/universalmotif_df.R

```diff
@@ -141,12 +141,27 @@ update_motifs <- function(motif_df, extrainfo = FALSE) {
   cols_new <- cols_new[cols_new != "motif"]
   cols_old <- colnames(old_df)
   cols_old <- cols_old[cols_old != "motif"]
+  extrainfo_holdout_cols <- NA_character_
   if (extrainfo) {
     # for now, just always update extrainfo...
     cols_extrainfo <- cols_new[!cols_new %in% cols_old]
     cols_extrainfo <- cols_extrainfo[cols_extrainfo != "bkg"]
     if (length(cols_extrainfo)) {
-      extrainfo_new <- updated_df[, cols_extrainfo, drop = FALSE]
+      # Keep id_cols temporarily to ensure sort order is OK
+      # TODO: in future use a better unique identifier
+      id_cols <- c("name", "consensus")
+      extrainfo_new <- updated_df[, c(id_cols, cols_extrainfo), drop = FALSE]
+      
+      # TOOD: hold out unsupported datatypes
+      # Current "supported" types are "character", "numeric" and "integer"
+      # Should logical be held out or not?
+      extrainfo_types <- vapply(extrainfo_new, class, character(1))
```

Comment:
Can you use `typeof` instead of `class`? I think the former should be safer.

Reply:
If I use `typeof` this interprets factors as numeric, not factors, so they get converted to their integer values (the new unit tests catch this though, so there's at least the infrastructure to catch when it's fixed). I guess another option is to handle factors separately? I'm not sure how to handle this one, so I've left it as `class` for now.